Emerald Insight.js

```diff
@@ -150,8 +150,16 @@ function scrape(doc, url) {
 			// trying to correct for this
 			for (let i = 0; i < item.creators.length; i++) {
 				if (!item.creators[i].firstName && item.creators[i].lastName.includes(" ")) {
-					item.creators[i].firstName = item.creators[i].lastName.match(/^\w+\s+(.+)/)[1];
-					item.creators[i].lastName = item.creators[i].lastName.replace(/\s.+/, "");
+					var author = doc.querySelector("a[href*='" + item.creators[i].lastName.split(" ")[0] + "'][class='contrib-search']");
+					if(author) {
+						item.creators[i].firstName = text(author, "span[class=given-names]");
+						item.creators[i].lastName = text(author, "span[class=surname]");
+					}
+					else {
+						Z.debug("No span tag for " + item.creators[i].lastName);
+						item.creators[i].firstName = item.creators[i].lastName.split(" ")[0];
+						item.creators[i].lastName = item.creators[i].lastName.split(" ").slice(1).join(" ");
```

Comment:
Is there a way to select all remaining elements of an array in js other than with a slice? In python I use [1:] but I didn't find any equivalent

Reply:
`slice` is fine. But let us maybe define a variable for `item.creators[i].lastName.split(" ")` and use it instead in the condition (line 152) and subsequent places.